authzed/api/v1/permission_service.proto

```diff
@@ -115,27 +121,34 @@ message ReadRelationshipsResponse {
   Relationship relationship = 2;
 }
 
+// Precondition specifies how and the existence or absence of certain
+// relationships as expressed through the accompanying filter should affect
+// whether or not the operation proceeds.
+//
+// MUST_NOT_MATCH will fail the parent request if any relationships match the
+// relationships filter.
+// MUST_MATCH will fail the parent request if there are no
+// relationships that match the filter.
+message Precondition {
+  enum Operation {
+    OPERATION_UNSPECIFIED = 0;
+    OPERATION_MUST_NOT_MATCH = 1;
+    OPERATION_MUST_MATCH = 2;
+  }
+
+  Operation operation = 1;
```

Comment:
I am not convinced that the operation is necessary.
When would "must not match" be used? The precondition is basically supposed to be used to enable writes to function with compare-and-swap semantics, which only ever requires "must match" logic.

Reply:
Joey's example of when you'd want "must not match": the first migration being ran for a system -- you'd want to assert that no relationships with any values for a migration would exist.